python/styleguide.md

```diff
@@ -0,0 +1,108 @@
+# Python Code Style Guide
+
+
+Documentation
+-------------
+
+* We use doc strings for package, module, class and function documentation.
+    * A doc string is a string that is the first statement in a package, module, class or function.
+    * These strings can be extracted automatically through the __doc__ member of the object. 
+
+* We always use the three double-quote `"""` format for doc strings.
+    * This is defined in [PEP 257](https://www.python.org/dev/peps/pep-0257/)
+
+* A doc string should be organized as
+    * a summary line (one physical line),
+    * a blank line,
+    * the rest of the doc string.
+
+### On Block comments
+
+* Comment on complicated parts of the code
+    * If you're going to have to explain it at the next code review, you should comment it now.
+
+* Don't describe the code.
+
+
+### Example
+
+* Function or method comments: 
+```python
+def fetch_bigtable_rows(big_table, keys, other_silly_variable=None):
+    """
+    Fetches rows from a Bigtable.
+
+    Retrieves rows pertaining to the given keys from the Table instance
+    represented by big_table.  Silly things may happen if
+    other_silly_variable is not None.
+
+    Args:
+        big_table: An open Bigtable Table instance.
+        keys: A sequence of strings representing the key of each table row
+            to fetch.
+        other_silly_variable: Another optional variable, that has a much
+            longer name than the other args, and which does nothing.
+
+    Returns:
+        A dict mapping keys to the corresponding table row data
+        fetched. Each row is represented as a tuple of strings. For
+        example:
+
+        {'Serak': ('Rigel VII', 'Preparer'),
+         'Zim': ('Irk', 'Invader'),
+         'Lrrr': ('Omicron Persei 8', 'Emperor')}
+
+        If a key from the keys argument is missing from the dictionary,
+        then that row was not found in the table.
+
+    Raises:
+        IOError: An error occurred accessing the bigtable.Table object.
+    """
+    pass
+```
+
+* Class comments:
+```python
+class SampleClass(object):
+    """Summary of class here.
```

Comment:
Historically most of our Python code-bases used a docstring format like this (even for short comments):

```
"""    <--- triple quotes on their own line
The comment.
       <--- blank line
"""
```

It might make sense to continue that practice because consistency :rainbow:. At some point this was also written down in Confluence but I can't find the page at the moment. Then again it might make sense to just adopt something like PEP 257 so we don't have to come up with our own formatting rules :)